spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/support/GitCredentialsProviderFactory.java

```diff
@@ -92,13 +92,6 @@ else if (hasText(username)) {
 			this.logger.debug("Constructing UsernamePasswordCredentialsProvider for URI " + uri);
 			provider = new UsernamePasswordCredentialsProvider(username, password.toCharArray());
 		}
-		else if (hasText(username) && !hasText(passphrase)) {
```

Comment:
Maybe it shouldn't be removed and add `&& hasText(password)` to the branch above? That will fix a potential NPE there anyway.

Reply:
@spencergibb This logic is tasting if _passphrase_ has text not password.  I agree that we should add a null check for `password` above.  But we probably don't want to use a `hasText` check since having an empty string for password is valid (we have a test for that).

Reply:
right, null check works